cstring.h

```diff
@@ -363,6 +374,36 @@ int64_t stringFindFirstCharArr(TString s, const char *pattern) {
     return -1; 
 }
 
+int64_t stringToInt(TString s) {
+    int64_t sign = 1;
+    int i = 0;
+
+    if (s.data[0] == '-') {
+        if (s.size == 1) {
+            setError(ERR_INVALID_NUMBER_REPR);
+            return 0;
+        }
+        sign = -1;
+        i++;
+    }
+
+    int64_t val = 0;
+    for (; i < s.size; i++) {
+        int64_t digit = stringCharToInt(s.data[i]);
+        
+        if (val > (INT64_MAX - digit) / 10) {
+            setError(ERR_NUMBER_OVERFLOW);
+            return val;
```

Comment:
Such a nice code!
just change to `>=` and add test for `stringToInt(INT64_MAX)`

Reply:
it will probably overflow with INT64_MIN due to the asymmetric range but you may just leave TODO